taxconverter/__main__.py

```
@@ -46,13 +50,19 @@ def all_to_taxvamb(df: pd.DataFrame):
 def ncbi_lineage():
     begintime = time.time()
     logger.info("Loading NCBI lineage")
-    df_ncbi = pd.read_csv(NCBI_LINEAGE, quoting=csv.QUOTE_NONE)
-    df_ncbi[LINEAGE_COL] = df_ncbi[TAXA_LEVELS].apply(lambda x: x.str.cat(sep=';'), axis=1)
-    df_ncbi = df_ncbi[['tax_id', LINEAGE_COL]]
-    df_ncbi['tax_id'] = df_ncbi['tax_id'].astype(str)
+    df_ncbi = pd.read_csv(NCBI_LINEAGE, quoting=csv.QUOTE_NONE, sep='\t')
+    map_child_parent = {k: v for k, v in zip(df_ncbi[CHILD_ID].astype(str), df_ncbi[PARENT_ID].astype(str))}
     elapsed = round(time.time() - begintime, 2)
-    logger.info(f"Loaded NCBI lineage with {len(df_ncbi)} entries in {elapsed} seconds")
-    return df_ncbi
+    logger.info(f"Loaded NCBI lineage with {len(map_child_parent)} entries in {elapsed} seconds")
+    return map_child_parent
+
+
+def get_lineage(tax_id, map_child_parent):
+    lineage = []
+    while tax_id != '1':
+        lineage.append(tax_id)
+        tax_id = map_child_parent.get(tax_id, '1')
+    return ';'.join(lineage[::-1])
```

This function may have the same issue in that it fails silently. Consider the function `centrifuge_data`. In it, if Centrifuge creates a taxonomy label which is not in our NCBI dataset, it fetch `1` as its parent, and be interpreted as a new, domain level clade.
Instead, I suggest not having the `1` fallback in the `get` call here. Then, if it turns out to be unworkable in practice (maybe lots of clades returned by Centrifuge aren't actually in NCBI), we can reconsider.